backend/auth/firebase.go

```diff
@@ -15,36 +15,56 @@
 package auth
 
 import (
-	"context"
-	"firebase.google.com/go/v4/auth"
-	"log"
-	"net/http"
-	"strings"
-	"time"
+  "context"
+  "firebase.google.com/go/v4/auth"
+  "net/http"
+  "strings"
+  "time"
 )
 
 type FirebaseAuth struct {
 	FirebaseClient firebaseAuthClient
 	AuthTimeout    time.Duration
+	AuthDomain     *string
 }
 
 type firebaseAuthClient interface {
 	VerifyIDToken(ctx context.Context, idToken string) (*auth.Token, error)
 }
 
-func (auth FirebaseAuth) Authenticate(next http.HandlerFunc) http.HandlerFunc {
+func (auth FirebaseAuth) Authorize(next http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		ctx, cancel := context.WithTimeout(r.Context(), auth.AuthTimeout)
+		_, cancel := context.WithTimeout(r.Context(), auth.AuthTimeout)
```

Comment:
This line doesn't do anything any more. I think it needs to be moved into `Authenticate`. You will need to pass `r.Context()` into it. See below.

Reply:
done